### PR TITLE
Feat/yield

### PR DESCRIPTION
**Pull request by zgorizzo69**
run:

```
mkdir lib run:

```
mkdir lib 
```
we should have a `ubiquity/ubiquity-dollar/contracts/yield-aggregator/lib` folder
```
forge install foundry-rs/forge-stdfor
```
```
forge install openzeppelin/openzeppelin-contracts
```
```
forge build
```
```
we should have a `ubiquity/ubiquity-dollar/contracts/yield-aggregator/lib` folder
```
forge install foundry-rs/forge-stdfor
```
```
forge install openzeppelin/openzeppelin-contracts
```
```
forge build
```

### CODE REVIEW

**Comment by 0x4007** [comment]
@0xcodercrane I hope you two can coordinate on merging the Foundry work. Not sure how you want to handle this but I'll let you take the lead as I have never used Foundry!

@zgorizzo69 if this is still in progress I suggest that you convert this to a draft pull request.

**Review by 0x4007 — DISMISSED** [review]

File: contracts/yield-aggregator/contracts/ProxyYieldAggregator.sol [hunk 1]
```diff
@@ -0,0 +1,328 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.10;
+import "./utils/CollectableDust.sol";
+import "openzeppelin-contracts/contracts/security/Pausable.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/extensions/ERC4626.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IUbiquityAlgorithmicDollarManager.sol";
+import "./interfaces/IERC20Ubiquity.sol";
+import "forge-std/Test.sol";
```
Comment (cuddle-dev) [inline contracts/yield-aggregator/contracts/ProxyYieldAggregator.sol]: we don't need Test.sol in production.

File: contracts/yield-aggregator/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+To run a test cases for specific one: <br />
+```
+forge test -m  testExample  --fork-url https://eth-mainnet.alchemyapi.io/v2/YOURKEYHERE -vvvvv
+```
+for test containing string
+```
+forge test --mc ERC4626  --fork-url https://eth-mainnet.alchemyapi.io/v2/YOURKEYHERE -v
+forge test --mc Proxy  --fork-url https://eth-mainnet.alchemyapi.io/v2/YOURKEYHERE -vvvv
```
Comment (cuddle-dev) [inline contracts/yield-aggregator/README.md:40]: I think we should be able to run all the contracts tests in one script here

File: contracts/yield-aggregator/test/utils/mocks/MockERC20.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  ) ERC20(_name, _symbol) {}
+
+  function mint(address to, uint value) public virtual {
+    _mint(to, value);
+  }
+
+  function burn(address from, uint value) public virtual {
+    _burn(from, value);
+  }
+}
```
Comment (cuddle-dev) [inline contracts/yield-aggregator/test/utils/mocks/MockERC20.sol:20]: MockFiles looks good to me.

**Review by cuddle-dev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**